src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -283,12 +283,25 @@ export function SavedObjectProvider(Promise, Private, Notifier, confirmModalProm
         });
     };
 
+    const displayDuplicateTitleConfirmModal = () => {
+      const confirmMessage =
+        `A ${this.getDisplayName()} with the title '${this.title}' already exists. Would you like to save anyway?`;
+
+      return confirmModalPromise(confirmMessage, { confirmButtonText: `Save ${this.getDisplayName()}` })
+        .catch(() => Promise.reject(new Error(SAVE_DUPLICATE_REJECTED)));
+    };
+
     /**
      * Returns a promise that resolves to true if either the title is unique, or if the user confirmed they
      * wished to save the duplicate title.  Promise is rejected if the user rejects the confirmation.
      */
-    const warnIfDuplicateTitle = () => {
-      // Don't warn if the user isn't updating the title, otherwise that would become very annoying to have
+    const checkForDuplicateTitle = (isTitleDuplicateConfirmed, onTitleDuplicate) => {
```

Comment:
I think this comment could be updated now.  Maybe something like:

If `isTitleDuplicateConfirmed` is true, this is a noop and a resolved promise is returned. If it is not, then this checks to see if the title is unique. If it is, a resolved promise is returned. If it is not, then onTitleDuplicate is called (if given) and a rejected promise is returned. If onTitleDuplicate is not given then a basic confirmation modal is shown to the user, which then determines if a resolved promise (they confirmed) or a rejected promise (they canceled) is returned.

I dunno, that's not great, but there are so many different flows for this function it gets tough to shorten the message.  I'm not a big fan of passing in a parameter that causes the function to be a no-op (isTitleDuplicateConfirmed) - it complicates the flow. Plus this function is actually only called once.  Not a big deal though, I see it might be easier to leave it in here so you don't have to split out the .then() part of the caller.

Maybe just get rid of the comment if it's to complicated to succinctly say what it does?

Reply:
I think removing the comment is the best course of action. No point describing the code logic in a comment since the author can just read thru the logic. That way, if/when the logic changes, an outdated comment does not exist.

In the new flow, `this.save` is called twice when there is a duplicate title. The first time, `onTitleDuplicate` gets called. The second time, `isTitleDuplicateConfirmed` is true so the save goes thru. The old flow avoided calling `this.save` twice by including the dialog in the flow of the original save call.